### PR TITLE
fix: Mount dir to kaniko using Minio

### PR DESCRIPTION
**Pull request by mojtaba-esk**
Currently, building from a local directory using `kaniko` builder is only possible when the directory is less than 3MB. The main reason is because we use config maps and try to make an archive of the current directory and set it in the Job instructions and since k8s does not allow a request more than 3MB, it fails for most of the repos.

So this PR proposes a new change where it first create a Minio file sharing service in the cluster and then it makes a `tar.gz` archive of the directory and send it to Minio. 
Next, it downloads it from an `init` container into a volume, which is also mounted onto the `Kaniko` container. 

This way we do not need to modify the Kaniko container and maintain another code base, and we can build a local directory on k8s and it is seamless to the user.

### CODE REVIEW

File: pkg/minio/minio.go [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "minio"}},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Name:  DeploymentName,
+						Image: Image,
+						Env: []v1.EnvVar{
+							{Name: "MINIO_ACCESS_KEY", Value: accessKey},
+							{Name: "MINIO_SECRET_KEY", Value: secretKey},
```
Comment (Bidon15) [inline pkg/minio/minio.go:79]: Should be consts

File: pkg/minio/minio.go [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+
+	// Create Minio deployment
+	minioDeployment := &appsV1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DeploymentName,
+			Namespace: m.Namespace,
+		},
+		Spec: appsV1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "minio"},
```
Comment (Bidon15) [inline pkg/minio/minio.go:69]: should be consts

File: pkg/minio/minio.go [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"app": "minio"}},
```
Comment (Bidon15) [inline pkg/minio/minio.go:72]: same

File: pkg/minio/minio.go [hunk 4]
```diff
@@ -0,0 +1,418 @@
...
+			existingPV = &pv
+			break
+		}
+	}
+
+	if existingPV == nil {
+		// Create a simple PV if no existing PV is suitable
+		_, err = m.Clientset.CoreV1().PersistentVolumes().Create(ctx, &v1.PersistentVolume{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "minio-pv-",
```
Comment (Bidon15) [inline pkg/minio/minio.go:374]: should be a const prefix

File: pkg/minio/minio.go [hunk 5]
```diff
@@ -0,0 +1,418 @@
...
+			AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+			Resources: v1.VolumeResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceStorage: storageQt,
+				},
+			},
+		},
+	}
+
+	_, err = pvcClient.Create(ctx, pvc, metav1.CreateOptions{})
```
Comment (Bidon15) [inline pkg/minio/minio.go:411]: why are we not allowing options in signature func?

Comment (Bidon15) [reply pkg/minio/minio.go:411]: Ideally, we need to refactor this in the future to accomodate such use-cases

**Review by Bidon15 — CHANGES_REQUESTED** [review]

File: pkg/minio/minio.go (hunk 4 above)
Comment (mojtaba-esk) [reply pkg/minio/minio.go:374]: sure

File: pkg/minio/minio.go (hunk 2 above)
Comment (mojtaba-esk) [reply pkg/minio/minio.go:69]: ok

File: pkg/minio/minio.go (hunk 5 above)
Comment (mojtaba-esk) [reply pkg/minio/minio.go:411]: done

**Review by Bidon15 — APPROVED** [review]

**Pull request merged**